lib/hooks.js

```diff
@@ -119,8 +130,15 @@ var Hooks = {
     if (hookTypes[hookType] && hookTypes[hookType].sync) {
       hooks.forEach(function(hook) {
```

Comment:
Wouldn't it be simpler to just append to the `hooks` array if for the proxy? Rather than add new logic